### PR TITLE
Pre-loading of import and export URLs

### LINKED ISSUE #156: Pre-loading of import and export URLs
Currently, users have to manually key in the URLs for the import and export of data. This is error-prone and takes time. We should consider pre-loading the export and import URLs into the settings page of the STEPS apk.

### PR DESCRIPTION
**Pull request by owais-vd**
close #156

### CODE REVIEW

File: src/main/java/com/onaio/steps/Properties.java [hunk 1]
```diff
@@ -31,16 +26,21 @@
...
+import java.util.HashMap;
+import java.util.Map;
 
 public class Properties {
-    private static final Map<String, String> props = new HashMap<String, String>();
+    private static final Map<String, String> props = new HashMap<>();
 
     static {
-        props.put(ENDPOINT_URL, "");
+        props.put(ENDPOINT_URL, "https://preview.steps.ona.io/upload-file");
```
Comment (ekigamba) [inline src/main/java/com/onaio/steps/Properties.java:39]: Is the the final production URL?

**Review by ekigamba — CHANGES_REQUESTED** [review]

**Comment by ekigamba** [comment]
The app crashed when I tried to open the Settings page even though I had ODK Collect installed. I had to open ODK Collect briefly so that it doesn't crash again. 

```
2022-04-01 17:45:09.499 3534-3534/com.onaio.steps E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.onaio.steps, PID: 3534
    java.lang.RuntimeException: Unable to resume activity {com.onaio.steps/com.onaio.steps.activities.SettingsActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void org.odk.collect.android.injection.config.AppDependencyComponent.inject(org.odk.collect.android.external.FormsProvider)' on a null object reference
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3784)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3816)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void org.odk.collect.android.injection.config.AppDependencyComponent.inject(org.odk.collect.android.external.FormsProvider)' on a null object reference
        at android.os.Parcel.createException(Parcel.java:1956)
        at android.os.Parcel.readException(Parcel.java:1918)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
        at android.content.ContentProviderProxy.query(ContentProviderNative.java:418)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:163)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:146)
        at android.content.ContentProviderClient.query(ContentProviderClient.java:136)
        at com.onaio.steps.model.ODKForm.ODKBlankForm.get(ODKBlankForm.java:76)
        at com.onaio.steps.activities.SettingsActivity.prepareAvailableFormList(SettingsActivity.java:210)
        at com.onaio.steps.activities.SettingsActivity.onResume(SettingsActivity.java:69)
        at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1412)
        at android.app.Activity.performResume(Activity.java:7292)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3776)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3816) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:193) 
        at android.app.ActivityThread.main(ActivityThread.java:6669) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
```

**Comment by owais-vd** [comment]
> The app crashed when I tried to open the Settings page even though I had ODK Collect installed. I had to open ODK Collect briefly so that it doesn't crash again.
> 
> ```
> 2022-04-01 17:45:09.499 3534-3534/com.onaio.steps E/AndroidRuntime: FATAL EXCEPTION: main
>     Process: com.onaio.steps, PID: 3534
>     java.lang.RuntimeException: Unable to resume activity {com.onaio.steps/com.onaio.steps.activities.SettingsActivity}: java.lang.NullPointerException: Attempt to invoke interface method 'void org.odk.collect.android.injection.config.AppDependencyComponent.inject(org.odk.collect.android.external.FormsProvider)' on a null object reference
>         at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3784)
>         at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3816)
>         at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51)
>         at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145)
>         at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
>         at android.os.Handler.dispatchMessage(Handler.java:106)
>         at android.os.Looper.loop(Looper.java:193)
>         at android.app.ActivityThread.main(ActivityThread.java:6669)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
>      Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void org.odk.collect.android.injection.config.AppDependencyComponent.inject(org.odk.collect.android.external.FormsProvider)' on a null object reference
>         at android.os.Parcel.createException(Parcel.java:1956)
>         at android.os.Parcel.readException(Parcel.java:1918)
>         at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:183)
>         at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
>         at android.content.ContentProviderProxy.query(ContentProviderNative.java:418)
>         at android.content.ContentProviderClient.query(ContentProviderClient.java:163)
>         at android.content.ContentProviderClient.query(ContentProviderClient.java:146)
>         at android.content.ContentProviderClient.query(ContentProviderClient.java:136)
>         at com.onaio.steps.model.ODKForm.ODKBlankForm.get(ODKBlankForm.java:76)
>         at com.onaio.steps.activities.SettingsActivity.prepareAvailableFormList(SettingsActivity.java:210)
>         at com.onaio.steps.activities.SettingsActivity.onResume(SettingsActivity.java:69)
>         at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1412)
>         at android.app.Activity.performResume(Activity.java:7292)
>         at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3776)
>         at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3816) 
>         at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:51) 
>         at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:145) 
>         at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70) 
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808) 
>         at android.os.Handler.dispatchMessage(Handler.java:106) 
>         at android.os.Looper.loop(Looper.java:193) 
>         at android.app.ActivityThread.main(ActivityThread.java:6669) 
>         at java.lang.reflect.Method.invoke(Native Method) 
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493) 
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858) 
> ```

the above crash is known I'll create a ticket separately

**Comment by owais-vd** [comment]
Tracking here #161 the above crash

File: src/main/java/com/onaio/steps/Properties.java (hunk 1 above)
Comment (owais-vd) [reply src/main/java/com/onaio/steps/Properties.java:39]: I don't think so but @ciremusyoka can confirm. I will also move the pre-load URL in `build.gradle` file.

Comment (ciremusyoka) [reply src/main/java/com/onaio/steps/Properties.java]: @owais-vd this is not the production WHO Qatar URL. This is https://swa.moph.gov.qa

Comment (owais-vd) [reply src/main/java/com/onaio/steps/Properties.java]: Cool, I'll update the URL for debug and release build.

**For Debug:** https://preview.steps.ona.io

**For Release:** https://swa.moph.gov.qa

**Review by hilpitome — APPROVED** [review]
looks good to me

File: src/main/java/com/onaio/steps/Properties.java (hunk 1 above)
Comment (ekigamba) [reply src/main/java/com/onaio/steps/Properties.java]: No maintain the current URLs as they are on the Play Store version. Other will enter the correct URL on their apps

Comment (owais-vd) [reply src/main/java/com/onaio/steps/Properties.java]: i've maintained the current release URL based on this https://github.com/onaio/steps-app/pull/160#discussion_r841653629

Comment (owais-vd) [reply src/main/java/com/onaio/steps/Properties.java]: @ciremusyoka needs your input on https://github.com/onaio/steps-app/pull/160#discussion_r848491252

Comment (owais-vd) [reply src/main/java/com/onaio/steps/Properties.java]: @bkimondiu needs your input on https://github.com/onaio/steps-app/pull/160#discussion_r848491252

Comment (bkimondiu) [reply src/main/java/com/onaio/steps/Properties.java]: @owais-vd I agree with @ekigamba that we shouldn't use MOPH Qatar's URLs as the defaults on the app. Could we retain the URLs as they are and later change them to point to the instance that WHO Global will deploy? cc: @ukanga

Comment (owais-vd) [reply src/main/java/com/onaio/steps/Properties.java]: ok, currently we are using https://preview.steps.ona.io.

**Pull request merged**